taui/src/components/route-card.js

```diff
@@ -16,45 +18,74 @@ export default class RouteCard extends React.PureComponent<Props> {
   render () {
     const {
       cardColor,
-      children,
       isFavorite,
       goToDetails,
       neighborhood,
+      origin,
       setActiveNeighborhood,
       setFavorite,
       title,
       userProfile
     } = this.props
 
+    const markerClass = `neighborhood-summary__marker ${neighborhood.active ? 'neighborhood-summary__marker--on' : ''}`
+    const { time } = neighborhood
+    const originLabel = origin ? origin.label || '' : ''
+    const currentDestination = userProfile.destinations.find(d => d.location.label === originLabel)
+
+    const modeKey = neighborhood.segments && neighborhood.segments.length
```

Comment:
Referencing `segments` isn't a reliable way to determine the user profile mode; unreachable places in transit mode have no segments. It would be better to check `hasVehicle` on `userProfile`.